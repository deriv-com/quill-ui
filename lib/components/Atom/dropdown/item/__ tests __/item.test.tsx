import React from "react";
import { render } from "@testing-library/react";
import DropdownItem from "..";

describe("DropdownItem component", () => {
    it("renders correctly", () => {
        const label = "Test Label";
        const { getByText } = render(<DropdownItem label={label} />);
        const renderedLabel = getByText(label);
        expect(renderedLabel).toBeInTheDocument();
    });

    it("applies the 'disabled' class when disabled prop is true", () => {
        const label = "Test Label";
        const { container } = render(<DropdownItem label={label} disabled />);
        const dropdownItem = container.querySelector(".quill-dropdown-item");
        expect(dropdownItem).toHaveClass(
            "quill-dropdown-item__selected--false__disabled--true",
        );
    });

    it("applies the 'selected' class when selected prop is true", () => {
        const label = "Test Label";
        const { container } = render(<DropdownItem label={label} selected />);
        const dropdownItem = container.querySelector(".quill-dropdown-item");
        expect(dropdownItem).toHaveClass(
            "quill-dropdown-item__selected--true__disabled--false",
        );
    });

    it("renders left icon when provided", () => {
        const label = "Test Label";
        const leftIcon = <span>Left Icon</span>;
        const { getByText } = render(
            <DropdownItem label={label} leftIcon={leftIcon} />,
        );
        const renderedLeftIcon = getByText("Left Icon");
        expect(renderedLeftIcon).toBeInTheDocument();
    });

    it("renders right icon when provided", () => {
        const label = "Test Label";
        const rightIcon = <span>Right Icon</span>;
        const { getByText } = render(
            <DropdownItem label={label} rightIcon={rightIcon} />,
        );
        const renderedRightIcon = getByText("Right Icon");
        expect(renderedRightIcon).toBeInTheDocument();
    });
});
