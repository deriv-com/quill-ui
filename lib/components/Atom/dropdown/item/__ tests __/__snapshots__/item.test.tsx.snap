// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`DropdownItem component renders checkbox selected when provided 1`] = `
<div>
  <li
    class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-left quill-dropdown-item-checkbox__selected--true__disabled--false"
  >
    <svg
      height="24"
      role="img"
      viewBox="0 0 32 32"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M9.75 7.75h12.5c1.367 0 2.5 1.133 2.5 2.5v12.5c0 1.406-1.133 2.5-2.5 2.5H9.75a2.47 2.47 0 0 1-2.5-2.5v-12.5c0-1.367 1.094-2.5 2.5-2.5m10.664 6.914h-.039c.39-.351.39-.937 0-1.328a.92.92 0 0 0-1.29 0L14.75 17.71l-1.836-1.836c-.39-.39-.976-.39-1.328 0a.856.856 0 0 0 0 1.29l2.5 2.5c.351.39.937.39 1.328 0z"
      />
    </svg>
    <p
      class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label"
    >
      Test Label
    </p>
  </li>
</div>
`;

exports[`DropdownItem component renders checkbox when provided 1`] = `
<div>
  <li
    class="quill-dropdown-item quill-dropdown-item__size--md quill-dropdown-item__align-left quill-dropdown-item-checkbox__selected--false__disabled--false"
  >
    <svg
      height="24"
      role="img"
      viewBox="0 0 32 32"
      width="24"
      xmlns="http://www.w3.org/2000/svg"
    >
      <path
        d="M22.25 9H9.75c-.703 0-1.25.586-1.25 1.25v12.5c0 .703.547 1.25 1.25 1.25h12.5c.664 0 1.25-.547 1.25-1.25v-12.5c0-.664-.586-1.25-1.25-1.25M9.75 7.75h12.5c1.367 0 2.5 1.133 2.5 2.5v12.5c0 1.406-1.133 2.5-2.5 2.5H9.75a2.47 2.47 0 0 1-2.5-2.5v-12.5c0-1.367 1.094-2.5 2.5-2.5"
      />
    </svg>
    <p
      class="quill-typography__body-text__size--md__weight--regular__decoration--default quill-typography__color--default quill-dropdown-item-label"
    >
      Test Label
    </p>
  </li>
</div>
`;
